src/Compilers/Core/CodeAnalysisTest/Analyzers/AnalyzerConfigTests.cs

```diff
@@ -932,6 +936,81 @@ public void EditorConfigToDiagnostics()
             }, options.Select(o => o.TreeOptions).ToArray());
         }
 
+        [Theory]
+        [InlineData("default")]
+        [InlineData("none")]
+        [InlineData("silent")]
+        [InlineData("refactoring")]
+        [InlineData("suggestion")]
+        [InlineData("warning")]
+        [InlineData("error")]
+        public void EditorConfigToDiagnosticsValidSeverity(string severity)
+        {
+            var expected = severity switch
+            {
+                "default" => ReportDiagnostic.Default,
+                "none" => ReportDiagnostic.Suppress,
+                "silent" => ReportDiagnostic.Hidden,
+                "refactoring" => ReportDiagnostic.Hidden,
+                "suggestion" => ReportDiagnostic.Info,
+                "warning" => ReportDiagnostic.Warn,
+                "error" => ReportDiagnostic.Error,
+                _ => throw ExceptionUtilities.UnexpectedValue(severity),
+            };
+
+            var configs = ArrayBuilder<AnalyzerConfig>.GetInstance();
+            configs.Add(Parse($@"
+[*.cs]
+dotnet_diagnostic.cs000.severity = {severity}
+", "/.editorconfig"));
+
+            var options = GetAnalyzerConfigOptions(["/test.cs"], configs);
+            configs.Free();
+
+            Assert.Equal([CreateImmutableDictionary(("cs000", expected))], Array.ConvertAll(options, o => o.TreeOptions));
+        }
+
+        [Theory]
+        [InlineData("unset")]
+        [InlineData("warn")]
+        [InlineData("")]
+        public void EditorConfigToDiagnosticsInvalidSeverity(string severity)
+        {
+            var configs = ArrayBuilder<AnalyzerConfig>.GetInstance();
+            configs.Add(Parse($@"
+[*.cs]
+dotnet_diagnostic.cs000.severity = {severity}
+", "/.editorconfig"));
+
+            var options = GetAnalyzerConfigOptions(["/test.cs"], configs);
+            configs.Free();
+
+            Assert.Equal([ImmutableDictionary<string, ReportDiagnostic>.Empty], Array.ConvertAll(options, o => o.TreeOptions));
+        }
+
+        /// <summary>
+        /// Verifies that the AnalyzerConfig parser ignores comment-like trailing text when parsing severity values.
+        /// This ensures the change from https://github.com/dotnet/roslyn/pull/51625 does not affect the parsing of
+        /// diagnostic severities.
+        /// </summary>
+        [Theory]
+        [InlineData("#")]
+        [InlineData(";")]
```

Comment:
I'd consider tweaking the inline data to add another case like:

```suggestion
        [InlineData("error # ignored text")]
        [InlineData("error ; ignored text")]
        [InlineData("warning ; ignored text")]
```

As otherwise this test could incorrectly pass if all parse errors were simply converted to ReportDiagnostic.Error.

Reply:
⏱️ I'll add a second line inside the test itself.